Doc/reference/datamodel.rst

```diff
@@ -1695,51 +1695,67 @@ Customizing module attribute access
 
 .. index::
    single: __getattr__ (module attribute)
+   single: __setattr__ (module attribute)
+   single: __delattr__ (module attribute)
    single: __dir__ (module attribute)
    single: __class__ (module attribute)
 
-Special names ``__getattr__`` and ``__dir__`` can be also used to customize
-access to module attributes. The ``__getattr__`` function at the module level
+Special names ``__getattr__``, ``__setattr__``, ``__delattr__`` and ``__dir__``
+can be also used to customize access to module attributes.
+
+The ``__getattr__`` function at the module level
 should accept one argument which is the name of an attribute and return the
 computed value or raise an :exc:`AttributeError`. If an attribute is
 not found on a module object through the normal lookup, i.e.
 :meth:`object.__getattribute__`, then ``__getattr__`` is searched in
 the module ``__dict__`` before raising an :exc:`AttributeError`. If found,
 it is called with the attribute name and the result is returned.
 
+The ``__setattr__`` function should accept two arguments, respectively, the
+name of an attribute and the value to be assigned and return :const:`None` or
+raise an :exc:`AttributeError`.  If present, this function overrides the
```

Comment:
It's surprising to say "return None", it's rare to write explicitly ``return None`` in Python. I suggest to omit it:

```suggestion
The ``__setattr__`` function should accept two arguments, respectively, the
name of an attribute and the value to be assigned. It can
raise an :exc:`AttributeError`.  If present, this function overrides the
```